service/endpoint_validation.go

```diff
@@ -97,8 +95,71 @@ func ValidateEndpointCreateRequest(dao dao.EndpointDao, ecr *model.EndpointCreat
 		return fmt.Errorf("the certificate authority cannot be empty")
 	}
 
-	if !util.SliceContainsString(validAvailabilityStatuses, ecr.AvailabilityStatus) {
-		return fmt.Errorf("invalid availability status")
+	// The team decided that the availability statuses will default to "in_progress" whenever they come empty, since
+	// setting them as "unavailable" by default may lead to some confusion to the calling clients.
+	if ecr.AvailabilityStatus == "" {
+		ecr.AvailabilityStatus = model.InProgress
+	} else {
+		if _, ok := model.ValidAvailabilityStatuses[ecr.AvailabilityStatus]; !ok {
```

Comment:
Should this be `model.ValidEndpointAvailabilityStatuses[ecr.AvailabilityStatus]`?

Reply:
Ooops, you're right. Thanks!